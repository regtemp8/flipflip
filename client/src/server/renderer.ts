import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
