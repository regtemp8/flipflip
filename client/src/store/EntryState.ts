export interface Identifiable {
  id: number
}

export interface EntryState<T extends Identifiable> {
  name: string
  nextID: number
  entries: Record<number, T>
}

export interface EntryUpdate<T> {
  id: number
  value: T
}

export function isEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
) {
  return state.entries[id] != null
}

export function setNextID<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (state.nextID <= entry.id) {
    state.nextID = entry.id + 1
  }
}

export function getEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
): T {
  return state.entries[id]
}

export function setEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  state.entries[entry.id] = entry
  setNextID(state, entry)
}

export function createEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (!isEntry(state, entry.id)) {
    state.entries[entry.id] = entry
    setNextID(state, entry)
  } else {
    throw new Error(
      `${state.name}: Create failed. Entry with ID ${entry.id} already exists.`
    )
  }
}

export function createEntries<T extends Identifiable>(
  state: EntryState<T>,
  entries: T[]
) {
  for (const entry of entries) {
    if (isEntry(state, entry.id)) {
      throw new Error(
        `${state.name}: Create failed. Entry with ID ${entry.id} already exists.`
      )
    }
  }

  entries.forEach((s) => (state.entries[s.id] = s))
  const maxID = Math.max(...entries.map((s) => s.id))
  setNextID(state, { id: maxID })
}

export function updateEntry<T extends Identifiable>(
  state: EntryState<T>,
  entry: T
) {
  if (isEntry(state, entry.id)) {
    state.entries[entry.id] = entry
  } else {
    throw new Error(
      `${state.name}: Update failed. Entry with ID ${entry.id} doesn't exist.`
    )
  }
}

export function deleteEntry<T extends Identifiable>(
  state: EntryState<T>,
  id: number
) {
  if (isEntry(state, id)) {
    delete state.entries[id]
  } else {
    throw new Error(
      `${state.name}: Delete failed. Entry with ID ${id} doesn't exist.`
    )
  }
}
