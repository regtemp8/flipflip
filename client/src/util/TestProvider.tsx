import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
