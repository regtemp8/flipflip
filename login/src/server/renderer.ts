import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
