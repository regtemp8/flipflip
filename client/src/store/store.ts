/// <reference path="../global.d.ts"/>
import { createReduxStore } from '../server/renderer'

const store = createReduxStore(
  window.flipflipAppStorage,
  window.flipflipConstants
)

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
