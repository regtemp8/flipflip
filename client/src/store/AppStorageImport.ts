import { initialApp } from "./app/data/App";
import { initialAudioState } from "./audio/slice";
import { initialCaptionScriptState } from "./captionScript/slice";
import { initialClipState } from "./clip/slice";
import { initialLibrarySourceState } from "./librarySource/slice";
import { initialOverlayState } from "./overlay/slice";
import { initialPlaylistState } from "./playlist/slice";
import { initialSceneState } from "./scene/slice";
import { initialSceneGridState } from "./sceneGrid/slice";
import { initialSceneGroupState } from "./sceneGroup/slice";
import { initialTagState } from "./tag/slice";
import { EntryState } from "./EntryState";
import type Tag from "./tag/Tag";
import type SceneGroup from "./sceneGroup/SceneGroup";
import type SceneGrid from "./sceneGrid/SceneGrid";
import type Scene from "./scene/Scene";
import type Playlist from "./playlist/Playlist";
import type Overlay from "./overlay/Overlay";
import type LibrarySource from "./librarySource/LibrarySource";
import type Clip from "./clip/Clip";
import type CaptionScript from "./captionScript/CaptionScript";
import type Audio from "./audio/Audio";
import type App from "./app/data/App";

export interface AppStorageImport {
  app: App;
  audio: EntryState<Audio>;
  captionScript: EntryState<CaptionScript>;
  clip: EntryState<Clip>;
  librarySource: EntryState<LibrarySource>;
  overlay: EntryState<Overlay>;
  playlist: EntryState<Playlist>;
  scene: EntryState<Scene>;
  sceneGrid: EntryState<SceneGrid>;
  sceneGroup: EntryState<SceneGroup>;
  tag: EntryState<Tag>;
}

export const initialAppStorageImport: AppStorageImport = {
  app: initialApp,
  audio: initialAudioState,
  captionScript: initialCaptionScriptState,
  clip: initialClipState,
  librarySource: initialLibrarySourceState,
  overlay: initialOverlayState,
  playlist: initialPlaylistState,
  scene: initialSceneState,
  sceneGrid: initialSceneGridState,
  sceneGroup: initialSceneGroupState,
  tag: initialTagState,
}