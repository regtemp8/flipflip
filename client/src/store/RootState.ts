import { RootState } from "./store";
import { initialApp } from "./app/data/App";
import { initialAudioState } from "./audio/slice";
import { initialCaptionScriptState } from "./captionScript/slice";
import { initialClipState } from "./clip/slice";
import { initialSystemConstants } from "./constants/SystemConstants";
import { initialLibrarySourceState } from "./librarySource/slice";
import { initialOverlayState } from "./overlay/slice";
import { initialPlaylistState } from "./playlist/slice";
import { initialSceneState } from "./scene/slice";
import { initialSceneGridState } from "./sceneGrid/slice";
import { initialSceneGroupState } from "./sceneGroup/slice";
import { initialTagState } from "./tag/slice";
import { initialPlayersState } from "./player/slice";
import { initialVideoClipperState } from "./videoClipper/slice";
import { initialScenePickerState } from "./scenePicker/slice";
import { initialCaptionScriptorState } from "./captionScriptor/slice";
import { initialSceneDetailState } from "./sceneDetail/slice";
import { initialAudioLibraryState } from "./audioLibrary/slice";
import { initialSourceScraperState } from "./sourceScraper/slice";

export const initialRootState: RootState = {
  app: initialApp,
  audio: initialAudioState,
  captionScript: initialCaptionScriptState,
  clip: initialClipState,
  constants: initialSystemConstants,
  librarySource: initialLibrarySourceState,
  overlay: initialOverlayState,
  playlist: initialPlaylistState,
  scene: initialSceneState,
  sceneGrid: initialSceneGridState,
  sceneGroup: initialSceneGroupState,
  tag: initialTagState,
  players: initialPlayersState,
  videoClipper: initialVideoClipperState,
  scenePicker: initialScenePickerState,
  captionScriptor: initialCaptionScriptorState,
  sceneDetail: initialSceneDetailState,
  audioLibrary: initialAudioLibraryState,
  sourceScraper: initialSourceScraperState
}