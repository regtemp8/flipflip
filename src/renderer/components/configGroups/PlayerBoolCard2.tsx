import React, { type ChangeEvent, useState } from 'react'

import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  Grid
} from '@mui/material'

import BaseSwitch from '../common/BaseSwitch'
import { useAppDispatch, useAppSelector } from '../../../store/hooks'
import { selectConstants } from '../../../store/constants/selectors'
import {
  setConfigGeneralSettingsPrioritizePerformance,
  setConfigGeneralSettingsConfirmSceneDeletion,
  setConfigGeneralSettingsConfirmBlacklist,
  setConfigGeneralSettingsConfirmFileDeletion,
  setConfigGeneralSettingsPortableMode,
  setConfigGeneralSettingsDisableLocalSave
} from '../../../store/app/slice'
import {
  selectAppConfigGeneralSettingsPrioritizePerformance,
  selectAppConfigGeneralSettingsConfirmSceneDeletion,
  selectAppConfigGeneralSettingsConfirmBlacklist,
  selectAppConfigGeneralSettingsConfirmFileDeletion,
  selectAppConfigGeneralSettingsPortableMode,
  selectAppConfigGeneralSettingsDisableLocalSave
} from '../../../store/app/selectors'
import { restoreAppStorageFromBackup } from '../../../store/app/thunks'

export default function PlayerBoolCard2() {
  const [portableDialog, setPortableDialog] = useState(false)

  const dispatch = useAppDispatch()
  const { portablePathExists, portablePath } = useAppSelector(selectConstants())
  const prioritizePerformance = useAppSelector(
    selectAppConfigGeneralSettingsPrioritizePerformance()
  )
  const portableMode = useAppSelector(
    selectAppConfigGeneralSettingsPortableMode()
  )

  const onTogglePortable = (
    e: ChangeEvent<HTMLInputElement>,
    checked: boolean
  ) => {
    if (checked && portablePathExists) {
      // Ask whether to keep local or keep portable
      onToggleDialog()
    } else {
      dispatch(setConfigGeneralSettingsPortableMode(checked))
    }
  }

  const onToggleDialog = () => {
    setPortableDialog(portableDialog)
  }

  const onChoosePortable = () => {
    dispatch(setConfigGeneralSettingsPortableMode(true))
    dispatch(restoreAppStorageFromBackup(portablePath))
    onToggleDialog()
  }

  return (
    <Grid container spacing={2} alignItems="center">
      <Grid item xs={12}>
        <BaseSwitch
          label={
            prioritizePerformance
              ? 'Prioritize Performance'
              : 'Prioritize Loading'
          }
          tooltip={
            <div>
              Prioritizing performance will smooth image effects, but may
              dramatically increase load times.
              <br />
              Prioritizing loading will decrease load times, but may result in
              jittery effects during playback
            </div>
          }
          selector={selectAppConfigGeneralSettingsPrioritizePerformance()}
          action={setConfigGeneralSettingsPrioritizePerformance}
        />
      </Grid>
      <Grid item xs={12}>
        <BaseSwitch
          label="Confirm Scene Deletion"
          tooltip="If disabled, no prompt will appear to confirm Scene deletion"
          selector={selectAppConfigGeneralSettingsConfirmSceneDeletion()}
          action={setConfigGeneralSettingsConfirmSceneDeletion}
        />
      </Grid>
      <Grid item xs={12}>
        <BaseSwitch
          label="Confirm Blacklist"
          tooltip="If disabled, no prompt will appear to confirm blacklisting a file"
          selector={selectAppConfigGeneralSettingsConfirmBlacklist()}
          action={setConfigGeneralSettingsConfirmBlacklist}
        />
      </Grid>
      <Grid item xs={12}>
        <BaseSwitch
          label="Confirm File Deletion"
          tooltip="If disabled, no prompt will appear to confirm File deletion"
          selector={selectAppConfigGeneralSettingsConfirmFileDeletion()}
          action={setConfigGeneralSettingsConfirmFileDeletion}
        />
      </Grid>
      <Grid item xs={12}>
        <BaseSwitch
          label="Portable Mode"
          tooltip="Portable Mode will save a copy of your data in the same directory as the FlipFlip executable, as well as the default save path. This needs to be enabled on each machine."
          selector={selectAppConfigGeneralSettingsPortableMode()}
          onChange={onTogglePortable}
        />
      </Grid>
      {portableMode && (
        <Grid item xs={12}>
          <BaseSwitch
            label="Disable Local Saves"
            tooltip="If on, data will only be saved in the same directory as the FlipFlip executable, and not at the default save path."
            selector={selectAppConfigGeneralSettingsDisableLocalSave()}
            action={setConfigGeneralSettingsDisableLocalSave}
          />
        </Grid>
      )}
      <Dialog
        open={portableDialog}
        onClose={onToggleDialog}
        aria-describedby="portable-description"
      >
        <DialogContent>
          <DialogContentText id="portable-description">
            Do you want to use the local data on this machine or existing
            portable data?
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={onToggleDialog} color="secondary">
            Local data
          </Button>
          <Button onClick={onChoosePortable} color="primary">
            Portable data
          </Button>
        </DialogActions>
      </Dialog>
    </Grid>
  )
}

;(PlayerBoolCard2 as any).displayName = 'PlayerBoolCard2'
