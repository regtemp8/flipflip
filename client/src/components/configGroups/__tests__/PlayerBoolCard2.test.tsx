import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import PlayerBoolCard2 from "../PlayerBoolCard2";
import TestProvider from "../../../util/TestProvider";
import store from "../../../store/store";

describe("PlayerBoolCard2", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <TestProvider store={store}>
        <PlayerBoolCard2 />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
