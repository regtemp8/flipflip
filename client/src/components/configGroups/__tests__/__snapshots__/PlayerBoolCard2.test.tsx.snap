// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerBoolCard2 should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <label
      aria-label={null}
      aria-labelledby={null}
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Prioritize Performance
      </span>
    </label>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <label
      aria-label="If disabled, no prompt will appear to confirm Scene deletion"
      aria-labelledby={null}
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Confirm Scene Deletion
      </span>
    </label>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <label
      aria-label="If disabled, no prompt will appear to confirm blacklisting a file"
      aria-labelledby={null}
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Confirm Blacklist
      </span>
    </label>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <label
      aria-label="If disabled, no prompt will appear to confirm File deletion"
      aria-labelledby={null}
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Confirm File Deletion
      </span>
    </label>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <label
      aria-label="Portable Mode will save a copy of your data in the same directory as the FlipFlip executable, as well as the default save path. This needs to be enabled on each machine."
      aria-labelledby={null}
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Portable Mode
      </span>
    </label>
  </div>
</div>
`;
